### PR TITLE
Fix Javascript error when filtering grid with a large dataset 

### PR DESCRIPTION
**Pull request by jcgueriaud1**
Here is a change for the grid to reset the _vidxOffset when the _effectiveSize is less than virtual count.

It should fix this ticket:
https://github.com/vaadin/vaadin-grid-flow/issues/1053

### CODE REVIEW

**Comment by jcgueriaud1** [comment]
Any idea why travis build failed?

File: src/vaadin-grid-scroller.html [hunk 1]
```diff
@@ -270,6 +270,8 @@
         const oldOffset = this._vidxOffset;
         if (this._accessIronListAPI(() => this._maxScrollTop) && this._virtualCount < this._effectiveSize) {
           this._adjustVirtualIndexOffset(delta);
+        } else {
+            this._vidxOffset = 0;
```
Comment (web-padawan) [inline src/vaadin-grid-scroller.html:274]: Please change the indentation here, should be two spaces.

File: test/filtering-huge-grid.html [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            var response = [];
+            for (var i = 0; i <= params.pageSize; i++) {
+              var element = params.page * params.pageSize + i;
+              if (element < grid.size) {
+                response.push({value: 'item' + element});
+              }
+            }
+            callback(response);
+          };
+          expect(getCellContent(getFirstVisibleItem(grid)).textContent).to.contain('item0');
```
Comment (web-padawan) [inline test/filtering-huge-grid.html:94]: Why is `expect` used outside of the test below, in `beforeEach` block?

When I remove this line, I'm getting the following error:

```
Error: expected 'item0' to include 'item299975'
```

File: src/vaadin-grid-scroller.html (hunk 1 above)
Comment (jcgueriaud1) [reply src/vaadin-grid-scroller.html]: Should be ok now.

File: test/filtering-huge-grid.html (hunk 2 above)
Comment (jcgueriaud1) [reply test/filtering-huge-grid.html]: I removed it and changed the test. It was wrong (and causes the timeout).

**Comment by web-padawan** [comment]
I will to merge this into other branch to trigger SauceLabs tests in all browsers. Then we can continue from there.

**Comment by jcgueriaud1** [comment]
Ok. It's better to run in all browsers :)

**Pull request merged**